src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DefaultProjectSnapshot.cs

```diff
@@ -101,5 +108,20 @@ public bool HasChangesComparedTo(ProjectSnapshot original)
 
             return !object.Equals(Configuration, original.Configuration);
         }
+
+        public bool HasTagHelpersChanged(ProjectSnapshot original)
+        {
+            if (original == null)
+            {
+                throw new ArgumentNullException(nameof(original));
+            }
+
+            if (TagHelpers == null || original.TagHelpers == null)
```

Comment:
`TagHelpers` wont be able to be `null`. `original.TagHelpers` on the other hand might be in a far-off land 😉